drivers/entropy/entropy_sam.c

```diff
@@ -58,14 +58,7 @@ static int entropy_sam_wait_ready(Trng * const trng, uint32_t flags)
 		}
 
 		if ((flags & ENTROPY_BUSYWAIT) == 0U) {
-			/* This internal function is used by both get_entropy,
-			 * and get_entropy_isr APIs. The later may call this
-			 * function with the ENTROPY_BUSYWAIT flag set. In
-			 * that case make no assumption that the kernel is
-			 * initialized when the function is called; so, just
-			 * do busy-wait for the random data to be ready.
-			 */
-			k_yield();
+			return -ENODATA;
```

Comment:
The ATMEL maintainer may want to comment, but I think you mixed what is happening here.
The isr version seems to **never** call with `(flags & ENTROPY_BUSYWAIT) == 0` into here, and therefore this path will never be taken from ISR.

With your change to line 106 this `if` is never true in any case; and the code will busy wait instead of yielding if not `_ready` (when called in thread mode)

Reply:
hmm, good catch. Actually the thing that needs to be changed here is just remove this k_yield() since the expected behavior is to block.